protocols/directdefund/directdefund.go

```diff
@@ -333,7 +331,26 @@ func (o *Objective) clone() Objective {
 
 // ObjectiveRequest represents a request to create a new direct defund objective.
 type ObjectiveRequest struct {
-	ChannelId types.Destination
+	ChannelId        types.Destination
+	objectiveStarted chan struct{}
+}
+
+// NewObjectiveRequest creates a new ObjectiveRequest.
+func NewObjectiveRequest(channelId types.Destination) ObjectiveRequest {
+	return ObjectiveRequest{
+		ChannelId:        channelId,
+		objectiveStarted: make(chan struct{}),
+	}
+}
+
+// SignalObjectiveStarted is used by the engine to signal the objective has been started.
+func (r ObjectiveRequest) SignalObjectiveStarted() {
+	close(r.objectiveStarted)
```

Comment:
Instead of just sending a `struct{}` to the `chan` we can close it to indicate the chan is [complete](https://gobyexample.com/closing-channels).